plugins/repoquery.py

```diff
@@ -222,6 +230,20 @@ def configure(self, args):
         demands.sack_activation = True
         demands.available_repos = True
 
+    def by_all_deps(self, name, query):
+        defaultquery = query.filter(name=name)
+        allpkgs = set()
+        requiresquery = self.by_requires(self.base.sack, name, query)
+        for reqpkg in requiresquery.run():
+            allpkgs.add(reqpkg)
+        for pkg in defaultquery.run():
+            for provide in pkg.provides:
+                provideQuery = query.filter(requires=provide)
```

Comment:
If I could have a wish, please don't use mixedCase notation for names of variables. We basically follow the (PEP 0008)[https://www.python.org/dev/peps/pep-0008/] which says:

> [...] names should be lowercase, with words separated by underscores as necessary to improve readability.